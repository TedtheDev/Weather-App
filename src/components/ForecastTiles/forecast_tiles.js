import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import ForecastTile from './ForecastTile/forecast_tile';

const ForecastTilesDiv = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    flex-wrap: wrap;
    overflow-y: auto;
`;

const renderTiles = (data) => {
    return data.map(day => <ForecastTile key={day.dt} day={day}/>);
}

const ForecastTiles = props => {
    if(props.loading) {
        return (
            <div>Loading</div>
        )
    }
    
    return (
        <ForecastTilesDiv>
            {renderTiles(props.data)}
        </ForecastTilesDiv>
    )
};

ForecastTiles.propTypes = {
    loading: PropTypes.bool,
    data: PropTypes.array
}

export default ForecastTiles;