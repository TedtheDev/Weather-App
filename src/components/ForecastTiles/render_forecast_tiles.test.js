import React from 'react';
import ForecastTiles from './forecast_tiles';
import renderer from 'react-test-renderer';

describe('<ForecastTiles />', () => {
    test('renders correctly', () => {
        const forecastData = [
            {"dt":1523782800,"temp":{"day":55.4,"min":47.86,"max":55.4,"night":47.86,"eve":55.4,"morn":55.4},"pressure":1014.51,"humidity":48,"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"speed":6.31,"deg":278,"clouds":12},
            {"dt":1523869200,"temp":{"day":59.58,"min":45.43,"max":59.88,"night":50.68,"eve":59.65,"morn":45.43},"pressure":1012.84,"humidity":77,"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"speed":4.38,"deg":273,"clouds":68,"rain":2.6},
            {"dt":1523955600,"temp":{"day":57.7,"min":44.82,"max":57.7,"night":54.39,"eve":56.71,"morn":44.82},"pressure":1007.91,"humidity":94,"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"speed":6.24,"deg":171,"clouds":92,"rain":8.58},
            {"dt":1524042000,"temp":{"day":48.18,"min":44.96,"max":51.28,"night":44.96,"eve":47.73,"morn":51.28},"pressure":1002.01,"humidity":93,"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"speed":15.35,"deg":337,"clouds":68,"rain":1.44},
            {"dt":1524128400,"temp":{"day":48.61,"min":42.12,"max":48.61,"night":42.12,"eve":46.38,"morn":48.61},"pressure":1000.05,"humidity":0,"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"speed":23.11,"deg":328,"clouds":73,"rain":3.89}
        ];
        
        const component = renderer.create(<ForecastTiles loading={false} data={forecastData}/>);

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});