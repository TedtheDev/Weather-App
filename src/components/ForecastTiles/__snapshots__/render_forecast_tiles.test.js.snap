// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastTiles /> renders correctly 1`] = `
<div
  className="sc-bwzfXH jYJRKU"
>
  <div
    className="sc-bdVaJa fEvWiI"
  >
    <div>
      Temp
    </div>
    <div>
      Date: 
      Sun Apr 15 2018 04:00:00 GMT-0500 (Central Daylight Time)
    </div>
    <div>
      Day: 
      55.4
    </div>
    <div>
      Evening: 
      55.4
    </div>
    <div>
      Max: 
      55.4
    </div>
    <div>
      Min: 
      47.86
    </div>
    <div>
      Morning: 
      55.4
    </div>
    <div>
      Night: 
      47.86
    </div>
  </div>
  <div
    className="sc-bdVaJa fEvWiI"
  >
    <div>
      Temp
    </div>
    <div>
      Date: 
      Mon Apr 16 2018 04:00:00 GMT-0500 (Central Daylight Time)
    </div>
    <div>
      Day: 
      59.58
    </div>
    <div>
      Evening: 
      59.65
    </div>
    <div>
      Max: 
      59.88
    </div>
    <div>
      Min: 
      45.43
    </div>
    <div>
      Morning: 
      45.43
    </div>
    <div>
      Night: 
      50.68
    </div>
  </div>
  <div
    className="sc-bdVaJa fEvWiI"
  >
    <div>
      Temp
    </div>
    <div>
      Date: 
      Tue Apr 17 2018 04:00:00 GMT-0500 (Central Daylight Time)
    </div>
    <div>
      Day: 
      57.7
    </div>
    <div>
      Evening: 
      56.71
    </div>
    <div>
      Max: 
      57.7
    </div>
    <div>
      Min: 
      44.82
    </div>
    <div>
      Morning: 
      44.82
    </div>
    <div>
      Night: 
      54.39
    </div>
  </div>
  <div
    className="sc-bdVaJa fEvWiI"
  >
    <div>
      Temp
    </div>
    <div>
      Date: 
      Wed Apr 18 2018 04:00:00 GMT-0500 (Central Daylight Time)
    </div>
    <div>
      Day: 
      48.18
    </div>
    <div>
      Evening: 
      47.73
    </div>
    <div>
      Max: 
      51.28
    </div>
    <div>
      Min: 
      44.96
    </div>
    <div>
      Morning: 
      51.28
    </div>
    <div>
      Night: 
      44.96
    </div>
  </div>
  <div
    className="sc-bdVaJa fEvWiI"
  >
    <div>
      Temp
    </div>
    <div>
      Date: 
      Thu Apr 19 2018 04:00:00 GMT-0500 (Central Daylight Time)
    </div>
    <div>
      Day: 
      48.61
    </div>
    <div>
      Evening: 
      46.38
    </div>
    <div>
      Max: 
      48.61
    </div>
    <div>
      Min: 
      42.12
    </div>
    <div>
      Morning: 
      48.61
    </div>
    <div>
      Night: 
      42.12
    </div>
  </div>
</div>
`;
